amazon-qldb-shell: init at 2.0.0

terlar opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
This package is missing from nixpkgs and when working with Amazon QLDB, it is a handy tool to have.

Relevant links:
- https://docs.aws.amazon.com/qldb/latest/developerguide/data-shell.html
- https://github.com/awslabs/amazon-qldb-shell

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

terlar committed:
amazon-qldb-shell: init at 2.0.0

IvarWithoutBones commented on pkgs/development/tools/amazon-qldb-shell/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    $out/bin/qldb --help
+    $out/bin/qldb --version | grep 'qldb ${version}'
+  '';
```
You could use [`testVersion`](https://github.com/NixOS/nixpkgs/blob/1b1fe46bb1f2d86cd47353c8d921f6c09677b294/pkgs/build-support/trivial-builders.nix#L785) to do this :)

terlar replied:
Wasn’t aware of this one, will have a look and update as needed.

IvarWithoutBones commented on pkgs/development/tools/amazon-qldb-shell/default.nix:
```diff
@@ -0,0 +1,34 @@
+    $out/bin/qldb --version | grep 'qldb ${version}'
+  '';
+
+  meta = {
+    description = "An interface to send PartiQL statements to Amazon Quantum Ledger Database (QLDB)";
+    homepage = "https://github.com/awslabs/amazon-qldb-shell";
+    license = lib.licenses.asl20;
+    maintainers = [ lib.maintainers.terlar ];
+  };
```
```suggestion
  meta = with lib; {
    description = "An interface to send PartiQL statements to Amazon Quantum Ledger Database (QLDB)";
    homepage = "https://github.com/awslabs/amazon-qldb-shell";
    license = licenses.asl20;
    maintainers = [ maintainers.terlar ];
  };
```

terlar replied:
I was trying to avoid `with` due to performance implications. Not sure how much this affects the evaluation though, just remember seeing that across the whole nixpkgs repo `with` added some percentage of time spent on evaluation.

I could switch to use `with` in order to align with the conventions.

IvarWithoutBones replied:
There luckily isn't a noticable difference, `lib` isn't that big compared to nixpkgs.

terlar commented:
@IvarWithoutBones I have addressed your comments, however due to the usage of `testVersion` there are quite some more changes. I hope it is not too controversial changes. I did search through the usage and most of them used the package itself as an input which I didn't like that much, I saw one package used a let with self assignment. 

I went for the same approach but named it package in order to be able to inherit it and I think it makes sense as well. I also moved out the pname and version, since I now have a let anyways.

This is what felt nicest to me, but if it is not following the general style I could change to use the package as input instead and go back to `rec` for pname/version or perhaps even use it from the package input.

IvarWithoutBones commented:
> I went for the same approach but named it package in order to be able to inherit it and I think it makes sense as well. I also moved out the pname and version, since I now have a let anyways.

Ah, fair enough. Using `let` here seems fine to me, inheriting `package` and not depending on the top-level attribute name sound like a good idea. 

I would personally say it'd be best to move `pname` and `version` back inside of the derivation, since nothing else uses it and that's a bit more consistent with the rest of nixpkgs. I don't think there's any guidelines on this though, so I think this is fine as well   :+1:

IvarWithoutBones reviewed (approved):
Looks good to me, thanks for the contribution!

Result of `nixpkgs-review pr 149726` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>amazon-qldb-shell</li>
  </ul>
</details>
